### PR TITLE
[10.0][MIG] port sale_invoice_auto_deliver to 10.0

### PR DESCRIPTION
**Pull request by Cedric-Pigeon**
Migration unmerged PR https://github.com/OCA/sale-workflow/pull/280 on 8.0 to 10.0

### CODE REVIEW

**Comment by antoniocanovas** [comment]
Testing in #ocacodesprint
Everything works fine but the "done quantity" which is not updated after invoicing:
 
captura de pantalla 2017-05-30 a las 16 02 51

**Comment by Cedric-Pigeon** [comment]
@rafaelbn @antoniocanovas bug fixed and I extended tests.

**Comment by Cedric-Pigeon** [comment]
@rafaelbn Could you please remove the need fixing  as I fixed the bug

**Review by lmignon — APPROVED** [review]
@OCA/crm-sales-marketing-maintainers Is it possible to merge this one. It's in production from months...

**Comment by pedrobaeza** [comment]
Isn't this overlapping somehow with `sale_automatic_workflow`?

**Comment by Cedric-Pigeon** [comment]
@pedrobaeza Not really. This addons adds an option to confirm the deliver order in the same time you generate the invoice. It is not automatic, it is on demand through the standard invoicing wizard

**Comment by pedrobaeza** [comment]
But the same effect can be got with the other module, acting on sales order confirmation. The only difference is that you trigger everything together. Has to be in a different moment in time?

**Comment by Cedric-Pigeon** [comment]
Yes but the use case is different. They prepare the goods and we the customer comes to pick it up  they gave it to him. They generate the invoice to get payment and confirm the stock ongoing in a while.
i agree the endpoint is the same but the way to go is different

**Comment by pedrobaeza** [comment]
OK, then let's go ahead with this. Do you want to keep both commits?

**Comment by Cedric-Pigeon** [comment]
@pedrobaeza I squashed last commit. Thanks

File: sale_invoice_auto_deliver/README.rst [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+To install this module, you need to:
+
+* Click on install button
+
+Usage
+=====
+
+To use this module, you need to:
+
+* Go to ...
```
Comment (pedrobaeza) [inline sale_invoice_auto_deliver/README.rst]: There are no usage instructions. Remove this section if empty or move part of the main description that are usage instructions.

File: sale_invoice_auto_deliver/README.rst [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/167/8.0
```
Comment (pedrobaeza) [inline sale_invoice_auto_deliver/README.rst]: Incorrect link. Maybe you should switch to new README by fragments system.

File: sale_invoice_auto_deliver/README.rst [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+in order to deliver products at the same time than the invoice. This functionality can be useful in the case where products are already take out of the stock.
+
+When invoice sale order line it is possible to choose the option : "Create Invoice + Automatic Deliver" to deliver the product and invoice it.
+
+Consider as pre-requirement that all products must be available.
+
+Installation
+============
+
+To install this module, you need to:
```
Comment (pedrobaeza) [inline sale_invoice_auto_deliver/README.rst]: These instructions are useless, so better to remove them.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
Sorry for this late review, but when I was going to merge, I see an incorrect README. Can you please fix it for merging?

**Comment by Cedric-Pigeon** [comment]
@pedrobaeza You are absolutely right. I just reworked this part and switched to fragments system

**Comment by pedrobaeza** [comment]
One last thing to make Travis happy:
```
sale_invoice_auto_deliver/readme/USAGE.rst:6: [E7901(rst-syntax-error), ]  Definition list ends without a blank line; unexpected unindent.
 ```

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**